C++/Dynamic-Programming/knapsack_dp.cpp

```diff
@@ -0,0 +1,82 @@
+#include <iostream>
+#include <algorithm>
+#include<cstring>
+
+using namespace std;
+
+// ========================== Problem Statement ===========================
+
+/*
+You are given a set of items, each with a weight and a value. 
+You are also given a knapsack with a certain capacity. Your goal 
+is to maximize the total value of items that you can put in the 
+knapsack, subject to the constraint that the total weight of the 
+items must not exceed the capacity of the knapsack.
+
+For example, suppose you have the following set of items:
+
+Item	Value	Weight
+1	100	20
+2	180	30
+3	260	10
+4	310	5
+5	40	25
+6	535	15
+7	695	30
+8	860	45
+9	920	50
+10	220	10
+The capacity of the knapsack is 200. Your goal is to choose a subset 
+of these items such that the total value is maximized, and the total 
+weight is less than or equal to 200.
+
+The 0/1 in the name refers to the fact that each item must either be 
+included or excluded (as opposed to a fractional knapsack problem, 
+where you can include a fraction of an item).
+*/
+
+//===========================C++ Implementation================================
+
+
+// Function to find the maximum value that can be put in a knapsack of capacity W
+int knapsack(int W, int wt[], int val[], int n)
```

Comment:
This looks more like program from this website: _https://www.geeksforgeeks.org/0-1-knapsack-problem-dp-10/_

under sub-heading: **0-1 Knapsack Problem using dynamic programming(Space optimized)**

Please refer these websites instead of **plagiarising** the code and changing the variable name.